jbpm-test-coverage/src/test/java/org/jbpm/test/functional/casemgmt/WorkCaseTest.java

```diff
@@ -0,0 +1,198 @@
+import org.jbpm.casemgmt.CaseMgmtUtil;
+import org.jbpm.process.instance.impl.demo.SystemOutWorkItemHandler;
+import org.jbpm.test.JbpmTestCase;
+import org.jbpm.test.casemgmt.WorkCaseService;
+import org.jbpm.test.casemgmt.WorkCaseService.WorkCaseMilestones;
+import org.jbpm.test.wih.InformWorkItemHandler;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.manager.audit.AuditService;
+import org.kie.api.runtime.manager.audit.NodeInstanceLog;
+import org.kie.api.runtime.manager.audit.ProcessInstanceLog;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.api.task.TaskService;
+import org.kie.api.task.model.TaskSummary;
+
+import qa.tools.ikeeper.annotation.BZ;
+
+public class WorkCaseTest extends JbpmTestCase {
+
+    private KieSession kieSession;
+    private RuntimeEngine runtimeEngine;
+    private CaseMgmtService caseMgmtService;
+    private TaskService taskService;
+    private AuditService auditService;
+
+    private InformWorkItemHandler iwih;
+    private WorkCaseService wcs;
+
+    @Before
+    public void setup() {
+
+        addWorkItemHandler("Milestone", new SystemOutWorkItemHandler());
+        iwih = new InformWorkItemHandler();
+        addWorkItemHandler("Inform", iwih);
+
+        kieSession = createKSession("org/jbpm/test/functional/casemgmt/WorkCase.bpmn2");
+        runtimeEngine = getRuntimeEngine();
+        caseMgmtService = new CaseMgmtUtil(runtimeEngine);
+        iwih.setCaseMgmtService(caseMgmtService);
+        taskService = runtimeEngine.getTaskService();
+        auditService = getLogService();
+        wcs = new WorkCaseService(runtimeEngine);
+    }
+
+    @Test(timeout = 30000)
+    public void testInformingOfInterestedPeople() {
+
+        String responsible = "john";
+        long caseId = wcs.createWork("Integrate module X into project Y", "Module X working in the project Y by ...", responsible);
+        wcs.addInformedPerson(caseId, "mary");
+
+        wcs.informAbout(caseId, "my message");
+        Assertions.assertThat(iwih.getLastMessage()).isEqualTo("my message");
+        Assertions.assertThat(iwih.getLastMessageToWhom()).contains("mary");
+
+        wcs.milestone(caseId, WorkCaseMilestones.CLOSE);
+
+    }
+
+    @Test(timeout = 30000)
+    public void testProlongDueDate() {
+
+        String responsible = "john";
+        long caseId = wcs.createWork("Integrate module X into project Y", "Module X working in the project Y by ...", responsible);
+        wcs.prolongTimeoutTo(caseId, 15);
+
+        try {
+            Thread.sleep(11000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ProcessInstanceLog pil = auditService.findProcessInstance(caseId);
+        Assertions.assertThat(pil.getStatus()).isEqualTo(ProcessInstance.STATE_ACTIVE);
+
+        try {
+            Thread.sleep(5000);
```

Comment:
Could you explain what's happening here? The way the test is currently written, it will cause false negatives (either the process will complete too quickly (above assert fails) or the process will take too long (below assert fails)).